src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -167,6 +178,42 @@ public void postEvent(final String topicId, final String partitionId, final Stri
     }
 
     @Override
+    public void syncPostBatch(String topicId, List<BatchItem> batch) throws EventPublishingException {
+        for (BatchItem item : batch) {
+            final ProducerRecord<String, String> record = new ProducerRecord<>(topicId,
+                    toKafkaPartition(item.getPartition()), item.getPartition(), item.getEvent().toString());
+
+            item.setStep(EventPublishingStep.PUBLISHING);
+
+            try {
+                kafkaProducer.send(record, kafkaSendCallback(item));
+            } catch (InterruptException | SerializationException | BufferExhaustedException e) {
+                LOG.error("Error posting event to kafka", e);
+                item.setPublishingStatus(EventPublishingStatus.FAILED);
+                throw new EventPublishingException("Error publishing message to kafka", e);
+            }
+        }
+
+        // this is not the most precise way of waiting for the send to complete, since it will flush all
+        // other concurrent partitions in the accumulator.
+        kafkaProducer.flush();
```

Comment:
Your're right. I changed it here https://github.com/zalando/nakadi/pull/186/commits/8f4adc36f670555c5b1019b7285f34510dc144f4 :+1: